sunpy/visualization/animator/tests/test_basefuncanimator.py

```diff
@@ -29,15 +34,56 @@ def button_func1(*args, **kwargs):
                                                     (plt.figure(), True, [[button_func1], ["hi"]])))
 def test_base_func_init(fig, colorbar, buttons):
     data = np.random.random((3, 10, 10))
-    func = partial(update_plotval, data=data)
-    funcs = [func]
-    ranges = [(0, 3)]
+    func0 = partial(update_plotval, data=data)
+    func1 = partial(update_plotval, data=data*10)
+    funcs = [func0, func1]
+    ranges = [(0, 3), (4, 7)]
 
     tfa = FuncAnimatorTest(data, funcs, ranges, fig=fig, colorbar=colorbar,
                            button_func=buttons[0],
                            button_labels=buttons[1])
 
     tfa.label_slider(0, "hello")
+    assert tfa.sliders[0]._slider.label.get_text() == "hello"
+
+    tfa._set_active_slider(1)
+    assert tfa.active_slider == 1
+
+    fig = plt.figure()
+    event = mback.KeyEvent(name='key_press_event', canvas=fig.canvas, key='down')
+    tfa._key_press(event)
+    assert tfa.active_slider == 0
+
+    event = mback.KeyEvent(name='key_press_event', canvas=fig.canvas, key='up')
```

Comment:
Can you add similar tests for left, right and `p`  here?

Reply:
Will update the tests with them.